server/api/routes/company.js

```diff
@@ -1,23 +1,34 @@
 const express = require('express')
 const router = express.Router()
-const { getCompany } = require('../../data_access_layer/company')
+const { getUserByCompanyCode } = require('../../data_access_layer/company')
 
 router.get('/:code', async (req, res) => {
   try {
     const { code } = req.params
-    const company = await getCompany(code)
-    if (company == null) {
+    const user = await getUserByCompanyCode(code)
+    if (user == null) {
       throw {
         statusCode: 404,
-        message: `Company with ID: ${code} not found.`
+        message: `No user exists with the company code '${code}'.`
       }
-    } 
-    return res.status(200).send(company)
+    } else if(user['Email']){
+      return res.status(409).send({
+        statusCode: 409,
+        message: `User with the company code '${code}' has already registered.`
+      })
+    } else{
+      const data = {
+        '_record': user['_record'],
+        'Company Code': user['Company Code'],
+        'Full Name': user['Full Name']
+      }
+      return res.status(200).send(data)
+    }

```

Comment:
make this an else if (!user[Email]) {}

Then make an else {} that throws an exception like this:
```js
throw {
    statusCode: 404,
    message: `Resource could not be found`
}
```